### PR TITLE
chromium: update to 95.0.4638.69

### PR DESCRIPTION
**Pull request by msisov**
Release notes -
https://chromereleases.googleblog.com/2021/10/stable-channel-update-for-desktop_28.html

License changes:
---------------
Added licenses:
* net/third_party/mozilla_win/LICENSE
- https://chromium-review.googlesource.com/c/chromium/src/+/3052905

Removed licenses:
* third_party/google_android_play_core/LICENSE
- https://chromium-review.googlesource.com/c/chromium/src/+/3085922
---------------

Test-built with:
* chromium-x11
      * dunfell, clang (dunfell branch), MACHINE="intel-core2-32"
      * dunfell, clang (dunfell branch), MACHINE="intel-corei7-64"
      * dunfell, clang (dunfell branch), MACHINE="raspberrypi3"
      * hardknott, clang (hardknott branch), MACHINE="raspberrypi3"
      * honister, clang, MACHINE="intel-core2-32"
    
* chromium-ozone-wayland:
      * dunfell, clang  (dunfell branch), MACHINE="intel-corei7-64"
      * dunfell, clang (dunfell branch), MACHINE="qemuarm64"
      * dunfell, clang (dunfell branch), MACHINE="m3ulcb"
      * hardknott, clang (hardknott branch), MACHINE="raspberrypi3
      * honister, clang, MACHINE="intel-corei7-64"

### CODE REVIEW

**Comment by msisov** [comment]
I was unable to test chromium-wayland with dunfell as it always resulted in the following error (haven't found any related changes. Thus, if anybody knows, your help is appreciated) -

| ../../base/third_party/double_conversion/double-conversion/bignum.cc:28:10: fatal error: 'algorithm' file not found
| #include <algorithm>
|          ^~~~~~~~~~~

**Review by rakuco — COMMENTED** [review]
The chromium-ozone-wayland -> chromium-wayland was not communicated in the PR or commit message, and it's a big change. First the dunfell build failed because meta-clang was trying to append to the chromium-ozone-wayland recipe that no longer exists; if I comment out the respective lines from meta-clang's `conf/layers.conf`, `bitbake chromium-wayland` still fails with

```
ERROR: Nothing PROVIDES 'virtual/libc-native' (but virtual:native:/data/src/yocto/poky/meta-clang/recipes-devtools/clang/compiler-rt_git.bb DEPENDS on or otherwise requires it). Close matches:
  virtual/libgl-native
  virtual/librpc-native
  virtual/libx11-native
ERROR: Required build target 'chromium-wayland' has no buildable providers.
Missing or unbuildable dependency chain was: ['chromium-wayland', 'gn-native', 'compiler-rt-native', 'virtual/libc-native']
```

I guess the main point is that renaming the recipe is a big change that can break things for users and for other layers that change the existing recipe with `bbappend` files, and I'd rather do that separately and after some discussion, even if the current name is no longer the best one.

**Comment by kraj** [comment]
> I was unable to test chromium-wayland with dunfell as it has always resulted in the following error (haven't found any related changes. Thus, if anybody knows, your help is appreciated) -
> 
> | ../../base/third_party/double_conversion/double-conversion/bignum.cc:28:10: fatal error: 'algorithm' file not found | #include | ^~~~~~~~~~~

we have clang-10 in dunfell and algorithm does exist in libc++ STL in clang-10 release. So I wonder if we are still using libstdc++ at STL on dunfell ? An experiment would be to use `dunfell-clang12` branch and see if you get same error. If you do then it will be clear that STL choice is wrong

**Review by rakuco — COMMENTED** [review]
After renaming chromium-wayland to chromium-ozone-wayland again, I hit a build issue in dunfell with the native part that's fixed by https://quiche.googlesource.com/quiche/+/cde409004ec894d7651a73ba1986ad2efe3683b5

```
In file included from ../../net/third_party/quiche/src/http2/adapter/window_manager.cc:1:
../../net/third_party/quiche/src/http2/adapter/window_manager.h:20:30: error: use of undeclared identifier 'size_t'
  typedef std::function<void(size_t)> WindowUpdateListener;
                             ^
```

> I was unable to test chromium-wayland with dunfell as it has always resulted in the following error (haven't found any related changes. Thus, if anybody knows, your help is appreciated)

It _might_ be related to the chromium-wayland rename and the meta-clang stuff that is supposed to be applied to chromium-ozone-wayland.

After applying the patch above and renaming the recipe back to chromium-ozone-wayland, I've managed to finish a dunfell build with chromium-ozone-wayland and MACHINE="qemuarm64".

**Comment by rakuco** [comment]
For the record, the chromium-x11 build passed on honister with MACHINE="intel-core2-32".

**Comment by r1mikey** [comment]
I've patched the in to our tree locally and fixed up the trailing quote issue in the wrapper script.

Built successfully on corei7-64 (meta-intel)-based, Poky distro-based environment.  This is a Wayland-only build.

I've run this on three of our usecases, covering video playback, audio, direct touch input and GL rendering.  I see no regressions over the m94 releases.

**Comment by rakuco** [comment]
Thanks for testing!

> and fixed up the trailing quote issue in the wrapper script.

What issue? Can you post a review comment with what's wrong?

**Comment by msisov** [comment]
> After renaming chromium-wayland to chromium-ozone-wayland again, I hit a build issue in dunfell with the native part that's fixed by https://quiche.googlesource.com/quiche/+/cde409004ec894d7651a73ba1986ad2efe3683b5
> 
> ```
> In file included from ../../net/third_party/quiche/src/http2/adapter/window_manager.cc:1:
> ../../net/third_party/quiche/src/http2/adapter/window_manager.h:20:30: error: use of undeclared identifier 'size_t'
>   typedef std::function<void(size_t)> WindowUpdateListener;
>                              ^
> ```
> 
> > I was unable to test chromium-wayland with dunfell as it has always resulted in the following error (haven't found any related changes. Thus, if anybody knows, your help is appreciated)
> 
> It _might_ be related to the chromium-wayland rename and the meta-clang stuff that is supposed to be applied to chromium-ozone-wayland.
> 
> After applying the patch above and renaming the recipe back to chromium-ozone-wayland, I've managed to finish a dunfell build with chromium-ozone-wayland and MACHINE="qemuarm64".

you're correct. This issue was due to renaming. Now all the targets pass. Thanks!

**Comment by r1mikey** [comment]
> Thanks for testing!
> 
> > and fixed up the trailing quote issue in the wrapper script.
> 
> What issue? Can you post a review comment with what's wrong?

The last line of  meta-chromium/recipes-browser/chromium/files/0004-wrapper-extra-flags.patch - it removes the trailing quote from "$@".  This will patch cleanly, but will cause errors at runtime.

**Comment by r1mikey** [comment]
> @r1mikey Is that 0003-v8-qemu-wrapper.patch ? It applies cleanly for me.

See my reply to @rakuco - the error is in the last line of  meta-chromium/recipes-browser/chromium/files/0004-wrapper-extra-flags.patch - the patch removes the trailing quote from "$@". 

The patch will apply cleanly, but will cause errors at runtime.  I only noticed the problem because we patch this file locally so as to avoid bash.

File: meta-chromium/recipes-browser/chromium/files/0004-wrapper-extra-flags.patch [hunk 1]
```diff
@@ -25,4 +25,4 @@ index dbbeb27ea037a..03c7ac4a77154 100755
 +
  # Note: exec -a below is a bashism.
 -exec -a "$0" "$HERE/@@PROGNAME@@" "$@"
-+exec -a "$0" "$HERE/@@PROGNAME@@" ${CHROME_EXTRA_ARGS} "$@"
++exec -a "$0" "$HERE/@@PROGNAME@@" ${CHROME_EXTRA_ARGS} "$@
```
[the remark this answers is not in the document]
Comment (r1mikey) [reply meta-chromium/recipes-browser/chromium/files/0004-wrapper-extra-flags.patch:28]: @msisov @rakuco This is the syntax error.

**Comment by rakuco** [comment]
> Built successfully on corei7-64 (meta-intel)-based, Poky distro-based environment. This is a Wayland-only build.

Can you share which branch you were using as well? We should update the commit/PR message with the new combinations we've tested after this pull request was sent.

**Comment by r1mikey** [comment]
> > Built successfully on corei7-64 (meta-intel)-based, Poky distro-based environment. This is a Wayland-only build.
> 
> Can you share which branch you were using as well? We should update the commit/PR message with the new combinations we've tested after this pull request was sent.

Sorry, completely forgot that.  I'm on honister for all dependencies.

**Review by rakuco — CHANGES_REQUESTED** [review]
@msisov We're almost there. Can you:
* Fix the wrapper script bug that @r1mikey identified
* Mention in the commit and PR messages that we've also tested:
  * chromium-x11 on honister with MACHINE="intel-core2-32"
  * chromium-ozone-wayland on honister with MACHINE="intel-corei7-64"

**Comment by msisov** [comment]
> @msisov We're almost there. Can you:
> 
> * Fix the wrapper script bug that @r1mikey identified
> * Mention in the commit and PR messages that we've also tested:
>   
>   * chromium-x11 on honister with MACHINE="intel-core2-32"
>   * chromium-ozone-wayland on honister with MACHINE="intel-corei7-64"

done.

**Review by rakuco — APPROVED** [review]
lgtm, thank you and thanks @r1mikey for helping review and test this change!

**Pull request merged**